### DISCUSSION TITLE
why infinity request  ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [TanStack/query#5012] [Q&A]
I'm just using it. I'm not doing anything else. Right, 

```jsx

import { useWhyDidYouUpdate } from 'ahooks'
import { useQuery } from '@tanstack/react-query'

export const Page = ()=> {
 const { data } = useQuery(['currentUser'], () => axios.get('/currentUser'))

 useWhyDidYouUpdate('s1212', {
    editable,
  })
}

```

<img width="586" alt="image" src="https://user-images.githubusercontent.com/124666577/220285860-7ebc7bcf-5a7c-4e59-af01-db6bc16a3837.png">

    "@tanstack/react-query": "^4.19.1",

**Comment by louis-young** [comment]
Hi 👋 

Are you sure it's actually making infinite requests and not just updating as expected when the query changes? If you think there's a genuine issue here, please could provide a runnable reproduction?

Thanks 🙂
